### PR TITLE
Remove table SelectedServiceRecipients and its references

### PR DESCRIPTION
**Pull request by RahulJha-mastek**
Story [AB#13236](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/13236)
Story [AB#14555](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14555)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Ordering.Api/Models/Summary/OrderSummaryModel.cs [hunk 1]
```diff
@@ -25,7 +25,7 @@ public static OrderSummaryModel Create(Order order)
             if (order is null)
                 throw new ArgumentNullException(nameof(order));
 
-            int serviceRecipientsCount = order.SelectedServiceRecipients.Count;
+            int serviceRecipientsCount = order.OrderItems.Where(o => o.CatalogueItem.CatalogueItemType == CatalogueItemType.Solution).SelectMany(o => o.OrderItemRecipients).Count();
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Models/Summary/OrderSummaryModel.cs:28]: I think we can remove `serviceRecipientsCount` and the `SectionModel.ServiceRecipients` below (line 47).

File: src/NHSD.BuyingCatalogue.Ordering.Api/Models/ServiceRecipientModel.cs [hunk 2]
```diff
@@ -10,10 +10,10 @@ public ServiceRecipientModel()
...
         }
 
-        internal ServiceRecipientModel(SelectedServiceRecipient recipient)
+        internal ServiceRecipientModel(ServiceRecipient recipient)
         {
-            Name = recipient.Recipient.Name;
-            OdsCode = recipient.Recipient.OdsCode;
+            Name = recipient.Name;
+            OdsCode = recipient.OdsCode;
         }
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Ordering.Api/Models/ServiceRecipientModel.cs:17]: Can this constructor be removed?

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/ServiceRecipients GetAll.feature [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿Feature: display all of the selected service recipients for an order
+﻿Feature: display all of the service recipients for an order
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/ServiceRecipients GetAll.feature:1]: I think we should also update the description of the first scenario to `Scenario: get the service recipients from an existing order`.

**Review by bjss-chris — CHANGES_REQUESTED** [review]
Looks good. However, I think there may be a couple of extra things that we can remove.

File: tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/ServiceRecipients GetAll.feature (hunk 3 above)
Comment (RahulJha-mastek) [reply tests/NHSD.BuyingCatalogue.Ordering.Api.IntegrationTests/Features/ServiceRecipients GetAll.feature:1]: Updated Scenario description with PR [435](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/435)

File: src/NHSD.BuyingCatalogue.Ordering.Api/Models/ServiceRecipientModel.cs (hunk 2 above)
Comment (RahulJha-mastek) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Models/ServiceRecipientModel.cs:17]: Removed constructor with PR [435](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/435)

File: src/NHSD.BuyingCatalogue.Ordering.Api/Models/Summary/OrderSummaryModel.cs (hunk 1 above)
Comment (RahulJha-mastek) [reply src/NHSD.BuyingCatalogue.Ordering.Api/Models/Summary/OrderSummaryModel.cs:28]: Removed `serviceRecipientsCount` with PR [435](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/435)

**Comment by RahulJha-mastek** [comment]
All changes in this PR are pushed with PR [435](https://github.com/nhs-digital-gp-it-futures/BuyingCatalogueOrdering/pull/435)

**Pull request closed without merging**